crates/router/src/connector/airwallex/transformers.rs

```diff
@@ -247,23 +297,63 @@ impl<F, T>
             types::PaymentsResponseData,
         >,
     ) -> Result<Self, Self::Error> {
-        let redirection_data =
-            item.response
-                .next_action
-                .map(|response_url_data| services::RedirectForm {
+        let redirection_data = item
+            .response
+            .next_action
+            .clone()
+            .map(|response_url_data| -> Result<_, errors::ConnectorError> {
+                common_utils::fp_utils::when(
+                    matches!(
+                        (response_url_data.stage, item.data.status),
+                        (
+                            AirwallexNextActionStage::WaitingDeviceDataCollection,
+                            enums::AttemptStatus::DeviceDataCollectionPending
+                        ) | (
+                            AirwallexNextActionStage::WaitingUserInfoInput,
+                            enums::AttemptStatus::AuthenticationPending
+                        )
+                    ),
+                    || {
+                        Err(errors::ConnectorError::InvalidConnectorResponse {
+                            field_name: "3DS step already completed, failing the payment",
+                        })
```

Comment:
Return the error response which will update the error message and code in payment intent and gracefully fail the payment.